plugins/woocommerce/src/Database/Migrations/CustomOrderTable/CLIRunner.php

```diff
@@ -0,0 +1,520 @@
+	}
+
+	/**
+	 * Verify meta data as part of verifying the order object.
+	 *
+	 * @param array $order_ids Order IDs.
+	 * @param array $failed_ids Array for storing failed IDs.
+	 *
+	 * @return array Failed IDs with meta details.
+	 */
+	private function verify_meta_data( array $order_ids, array $failed_ids ) : array {
+		global $wpdb;
+		if ( ! count( $order_ids ) ) {
+			return array();
+		}
+		$excluded_columns             = $this->post_to_cot_migrator->get_migrated_meta_keys();
+		$excluded_columns_placeholder = implode( ', ', array_fill( 0, count( $excluded_columns ), '%s' ) );
+		$order_ids_placeholder        = implode( ', ', array_fill( 0, count( $order_ids ), '%d' ) );
+		$meta_table                   = OrdersTableDataStore::get_meta_table_name();
+
+		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared, WordPress.DB.PreparedSQL.NotPrepared -- table names are hardcoded, orders_ids and excluded_columns are prepared.
+		$query           = $wpdb->prepare(
+			"
+SELECT {$wpdb->postmeta}.post_id as entity_id, {$wpdb->postmeta}.meta_key, {$wpdb->postmeta}.meta_value
+FROM $wpdb->postmeta
+WHERE
+      {$wpdb->postmeta}.post_id in ( $order_ids_placeholder ) AND
+      {$wpdb->postmeta}.meta_key not in ( $excluded_columns_placeholder )
+ORDER BY {$wpdb->postmeta}.post_id ASC, {$wpdb->postmeta}.meta_key ASC;
+",
+			array_merge(
+				$order_ids,
+				$excluded_columns
+			)
+		);
+		$data_to_migrate = $wpdb->get_results( $query, ARRAY_A );
+		// phpcs:enable
+
+		$clubbed_data_to_migrate = $this->club_data( $data_to_migrate );
+
+		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared, WordPress.DB.PreparedSQL.NotPrepared -- table names are hardcoded, orders_ids and excluded_columns are prepared.
+		$migrated_query = $wpdb->prepare(
+			"
+SELECT $meta_table.order_id as entity_id, $meta_table.meta_key, $meta_table.meta_value
+FROM $meta_table
+WHERE
+	$meta_table.order_id in ( $order_ids_placeholder )
+ORDER BY $meta_table.order_id ASC, $meta_table.meta_key ASC;
+",
+			$order_ids
+		);
+		$migrated_data  = $wpdb->get_results( $migrated_query, ARRAY_A );
+		// phpcs:enable
+
+		$clubbed_migrated_data = $this->club_data( $migrated_data );
+
+		foreach ( $clubbed_data_to_migrate as $order_id => $meta ) {
+			foreach ( $meta as $meta_key => $values ) {
+				if ( ! isset( $clubbed_migrated_data[ $order_id ] ) || ! isset( $clubbed_migrated_data[ $order_id ][ $meta_key ] ) ) {
+					$diff = $values;
+				} else {
+					$diff = array_diff( $values, $clubbed_migrated_data[ $order_id ][ $meta_key ] );
+				}
+
+				if ( count( $diff ) ) {
+					if ( ! isset( $failed_ids[ $order_id ] ) ) {
+						$failed_ids[ $order_id ] = array();
+					}
+					$failed_ids[ $order_id ][] = array(
+						'order_id'   => $order_id,
+						'meta_key'   => $meta_key,
+						'meta_value' => $diff,
+					);
+				}
+			}
+		}
+
+		return $failed_ids;
+	}
+
+	/**
+	 * Helper method to club response from meta queries into order_id > meta_key.
```

Comment:
Sorry, can't help myself, is this some meaning of club I'm not familiar with? Maybe @barryhughes as a native speaker can advise if this is understandable 😀

Reply:
Well, not 100% sure ... but I'm assuming [clubbed ("forcibly beaten")](https://classroomclipart.com/images/gallery/Animations/History/caveman_with_club_animation_5A.gif) into shape, because we're ensuring the array has a certain structure even in the absence of various fields. 

*Could possibly also allude to [a type of chocolate biscuit from the 1980s](https://www.youtube.comwatch?v=8LmZCMmNPkM), but this seems less likely.*

Could be something to rename? Sidebar (may seem unrelated, but it came to mind because what if we want to change nomenclature further down the line, for other methods that are not private): should all of this be in the public namespace vs the internal namespace?

Reply:
Yeah, by `club` I mean the biscuit thingy here. As you can see, just like that chocolate biscuit, we have a layer of order_id and then between two order ids, we have data for failed meta values sandwiched between them.

Actually, let me change it to the sandwich for better readability.

Reply:
Or maybe I will change it to say `normalize`, will see.